packages/header-change-detection/lib/lambda/header-check.ts

```diff
@@ -0,0 +1,300 @@
+    // Then get all the security headers from each response
+    const headers: Headers = new Map() as Headers;
+
+    Object.entries(response.headers).forEach(([headerName, value]) => {
+      if (securityHeaders?.includes(headerName))
+        headers.set(headerName, value as string);
+    });
+
+    currentUrlHeaders.set(url, headers);
+  })));
+
+  return currentUrlHeaders;
+}
+
+/**
+ * Get values stored in DynamoDB table from a list of string keys.
+ * Assumes the call is made to the header change detection table.
+ * This table has a primary key of Url (string) with an unknown
+ * number of string fields.
+ *
+ * @param keys array of strings
+ */
+const getStoredValues = async (keys: string[]): Promise<URLHeaders> => {
+  if (keys.length === 0) {
+    console.log("No keys were passed")
+    return new Map();
+  }
+
+  // Construct the command input
+  const primaryKeys = keys.map(url => {
+    return {
+      Url: {
+        "S": url
+      }
+    }
+  })
+  const requestItems = {
+    [TABLE]: {
+      "Keys": primaryKeys
+    }
+  }
+
+  return dynamoBatchRequest(requestItems);
+}
+
+/**
+ * Update stored headers for the given url.
+ * If the header no longer has a value, delete it. Otherwise update it.
+ *
+ * @param url the url to update - this is the primary key
+ * @param headers record of headers to update
+ */
+const updateStoredValues = async (url: string, headers: Headers) => {
+  // Convert headers to attribute value update attributes
+  const attributes: Record<string, AttributeValueUpdate> = {};
+  headers.forEach((value, headerName) => {
+    // If the value exists update it, otherwise remove it
+    if (value) {
+      attributes[headerName] = {
+        Value: {
+          "S": value
+        },
+        Action: 'PUT'
+      }
+    } else {
+      attributes[headerName] = {
+        Action: 'DELETE'
+      }
+    }
+  });
+
+  if (Object.values(attributes).length === 0) {
+    console.log(`No attribute value changes for ${url}`);
+    return;
+  }
+
+  return dynamoUpdateRequest(url, attributes);
+}
+
+/**
+ * Recursive function to get multiple items from a DynamoDB table
+ *
+ * @param requestItems
+ * @returns Promise<URLHeaders>
+ */
+const dynamoBatchRequest = async (requestItems: Record<string, KeysAndAttributes> | undefined): Promise<URLHeaders> => {
+  console.log(`Starting batch request with items: ${JSON.stringify(requestItems)}`);
+
+  // Validate that request items has values
+  if (Object.keys(requestItems || {})?.length === 0) return new Map();
+
+  const batchGetInput: BatchGetItemCommandInput = { RequestItems: requestItems }
+  const batchGetCommand = new BatchGetItemCommand(batchGetInput);
+
+  // Fetch stored stored headers
+  const response = await DB_CLIENT.send(batchGetCommand);
+  const responses = response.Responses?.[TABLE]!;
+
+  console.log(`Got following data from dynamo table: ${JSON.stringify(responses)}`);
+
+  const storedUrlHeaders: URLHeaders = new Map() as URLHeaders;
```

Comment:
Same here, having to `as ...` this feels weird. TS would be upset that we haven't provided generics for the Map, so we could just provide those and we won't have to persuade it.

Reply:
Oh I didn't realise you could do that with a map. I'll change the others too